content/docs/command-reference/push.md

```diff
@@ -18,55 +18,44 @@ positional arguments:
-these commands.
+[metrics](/doc/command-reference/metrics), etc.) remotely are the most common
+use cases for these commands.
 
-The `dvc push` command allows us to upload data to remote storage. It doesn't
-save any changes to the code, `dvc.yaml`, or `.dvc` files (those should be saved
-with `git commit` and `git push`).
+The `dvc push` command uploads data to
+[remote storage](/doc/command-reference/remote). It doesn't save any changes to
+the code, `dvc.yaml`, or `.dvc` files (that should be saved with `git commit`
+and `git push`).
 
-💡 For convenience, a Git hook is available to automate running `dvc push` after
-`git push`. See `dvc install` for more details.
-
-Under the hood a few actions are taken:
+The default remote is used (see `dvc config core.remote`) unless the `--remote`
+option is used. See `dvc remote` for more information on how to configure a
+remote.
 
-- The push command by default uses all stages (in `dvc.yaml` and `dvc.lock`) and
-  `.dvc` files in the <abbr>workspace</abbr>. The command options will either
-  limit or expand the set of stages or `.dvc` files to consult.
+Without arguments, it uploads all files and directories missing from remote
+storage, found as <abbr>outputs</abbr> in the stages (in `dvc.lock`) or `.dvc`
+files present in the workspace (the `--all-branches` and `--all-tags` enable
+using multiple workspace versions).
 
-- For each <abbr>output</abbr> referenced in every selected stage or `.dvc`
-  file, DVC finds a corresponding file or directory in the <abbr>cache</abbr>.
-  DVC then checks whether it exists in the remote. From this, DVC gathers a list
-  of files missing from the remote storage.
-
-- Upload the cache files missing from remote storage, if any, to the remote.
+The `targets` given to this command (if any) limit what to push. It accepts
+paths to tracked files or directories (including paths inside tracked
+directories), `.dvc` files, and stage names (found in `dvc.yaml`).
 
-The DVC `push` command always works with a remote storage, and it is an error if
-none are specified on the command line nor in the configuration. The default
-remote is used (see `dvc config core.remote`) unless the `--remote` option is
-used. See `dvc remote` for more information on how to configure a remote.
+💡 For convenience, a Git hook is available to automate running `dvc push` after
+`git push`. See `dvc install` for more details.
 
-With no arguments, just `dvc push` or `dvc push --remote REMOTE`, it uploads
-only the files (or directories) that are new in the local repository to remote
-storage. It will not upload files associated with earlier commits in the
-<abbr>repository</abbr> (if using Git), nor will it upload files that have not
-changed.
+For all <abbr>outputs</abbr> referenced in each target, DVC finds the
+corresponding files and directories in the <abbr>cache</abbr> (identified by
+hash values saved in `dvc.yaml` and `.dvc` files). DVC then gathers a list of
```

Comment:
hash values are not saved in dvc.yaml

Reply:
Typo... Thanks for catching that one. Fixed.